xcom_proto/XcomLAN.py

```diff
@@ -13,13 +9,75 @@
 from .protocol import Package
 from .XcomAbs import XcomAbs
 
+##
+# Class abstracting Xcom-LAN TCP network protocol
+##
 
-class XcomLAN(XcomAbs):
+class XcomLANTCP(XcomAbs):
+    def __init__(self, serverIP: str, dstPort=4002, srcPort=4001, localPort=5000):
+        """
+        Package requests are being sent to serverIP : dstPort using TCP protocol.
 
-    def __init__(self, serverIP: str, dstPort=4002, srcPort=4001):
+        The srcPort is needed for the server to listen to the TCP responses locally.
```

Comment:
The Moxa operating mode is configured as in the following screen shot:
![image](https://github.com/zocker-160/xcom-protocol/assets/97175692/56e87faf-0b83-4f00-aa60-10301bf144b5)

Can you point to where Studer documents their TCP server socket on their portal, or the TCP client on the Moxa?  I have only seen their documentation with the Moxa configured to be TCP server.  Or should this documentation come from Moxa?

Reply:
The key issue here is that Moxa is set as TCP Client (which you need in order to connect to Studer portal), but this also means in order to make this implementation work, one needs to implement a TCP Server.

The code right now does expect the Moxa to connect to a local TCP socket, which IIRC it not do. It does that only for UDP, because for some reason it does not simply send a response package.